orchestrator/lib/coolfhir/import.go

```diff
@@ -0,0 +1,89 @@
+package coolfhir
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	fhirclient "github.com/SanteonNL/go-fhir-client"
+	"github.com/zorgbijjou/golang-fhir-models/fhir-models/fhir"
+	"io"
+	"net/http"
+	"os"
+	"slices"
+	"strings"
+)
+
+// ImportResources imports resources from the given URLs into the FHIR server.
+// The URLs must point to JSON documents that contain a valid FHIR Bundle resource.
+// The URLs can only contain resources of the given types, otherwise an error is returned.
+// If the Bundle was posted to the FHIR server, the HTTP status code is returned.
+func ImportResources(ctx context.Context, client fhirclient.Client, resourceTypes []string, urlToLoad string) error {
+	var bundleData []byte
+	fileUrlPrefix := "file://"
+	if strings.HasPrefix(urlToLoad, fileUrlPrefix) {
+		filePath := urlToLoad[len(fileUrlPrefix):]
+		var err error
+		bundleData, err = os.ReadFile(filePath)
+		if err != nil {
+			return err
+		}
+	} else {
+		httpRequest, err := http.NewRequestWithContext(ctx, http.MethodGet, urlToLoad, nil)
+		if err != nil {
+			return err
+		}
+		httpResponse, err := http.DefaultClient.Do(httpRequest)
+		if err != nil {
+			return err
+		}
+		defer httpResponse.Body.Close()
+		if httpResponse.StatusCode <= 199 || httpResponse.StatusCode >= 300 {
+			return fmt.Errorf("unexpected status code: %d", httpResponse.StatusCode)
+		}
+		bundleData, err = io.ReadAll(io.LimitReader(httpResponse.Body, 1024*1024*10+1)) // Limit to 10mb
+		if err != nil {
+			return err
+		}
+	}
+
+	// TODO: Might want to make a "secure" http.Client that always does this
```

Comment:
Further in the vein of security, do we maybe want to provide some kind of whitelist for URLs we can download resources from?

Reply:
I considered this;
- The configured URLs are part of deployment configuration so at least 2 people should be involved
- You should use HTTPS URLs to avoid a MITM changing stuff
- The risk is probably very small, only frustrating the process.

I limit the resources being loaded into the FHIR server to Questionnaire and HealthcareService resources (and at 10mb), so I think the risk is quite low.

Reply:
Great, happy with that. Nice consideration to make